packages/react-widgets/src/widgets/CategoryWidget.js

```diff
@@ -144,7 +148,11 @@ CategoryWidget.propTypes = {
   title: PropTypes.string.isRequired,
   dataSource: PropTypes.string.isRequired,
   column: PropTypes.string.isRequired,
-  operationColumn: PropTypes.string,
+  operationColumn: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.arrayOf(PropTypes.string)
+  ]),
+  joinOperation: PropTypes.oneOf(Object.values(AggregationTypes)),
```

Comment:
We could use a custom PropTypes validator on this prop and the same on similar components, to ensure it's defined when the column prop is multiple.

Something like:
```jsx
const columnAggregationOn = (columnPropName) => (props, propName) => {
  if (Array.isArray(props[columnPropName])) {
    if (!props[propName]) {
      return new Error(`Prop ${propName} must be defined if ${columnPropName} is an array of column`);
    }
    if (!Object.values(AggregationTypes)) {
      return new Error(`Prop ${propName} must be a valid aggregation operator`);
    }
  }
};

// and in proptypes declaration: 
component.propTypes = {
    // ...
    operationColumn: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
    joinOperation: columnAggregationOn('operationColumn'),
    // ...
}
```

It could also share code with `checkParams` as it's doing basically the same thing, but PropType validator shouldn't throw.

Reply:
I really like it. Thanks!

Reply:
I love this solution. Already changed @bbecquet 👌🏻